ygnmi/ygnmi.go

```diff
@@ -313,19 +313,23 @@ func (w *Watcher[T]) Await() (*Value[T], error) {
 // Calling Await on the returned Watcher waits for the subscription to complete.
 // It returns the last observed value and a boolean that indicates whether that value satisfies the predicate.
 func Watch[T any](ctx context.Context, c *Client, q SingletonQuery[T], pred func(*Value[T]) error, opts ...Option) *Watcher[T] {
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithCancel(ctx)
 	w := &Watcher[T]{
 		errCh: make(chan error, 1),
 	}
 
 	resolvedOpts := resolveOpts(opts)
 	sub, err := subscribe[T](ctx, c, q, gpb.SubscriptionList_STREAM, resolvedOpts)
 	if err != nil {
+		cancel()
```

Comment:
This doesn't hurt anything, but not necessary, right? When a streaming RPC returns a non-nil error, it necessarily ends. The only case we technically need it is when we return early without having received an error. If that's right, you could delay the WithCancel call until right before the receiveStream call, and just leave the "defer cancel()" where it is.

Reply:
this was mostly to keep the linter happy.